pkg/actions/addon/update.go

```diff
@@ -85,28 +86,28 @@ func (a *Manager) Update(addon *api.Addon, wait bool) error {
 	return nil
 }
 
-func (a *Manager) updateWithNewPolicies(addon *api.Addon) (string, error) {
+func (a *Manager) updateWithNewPolicies(ctx context.Context, addon *api.Addon) (string, error) {
 	stackName := a.makeAddonName(addon.Name)
-	stack, err := a.stackManager.DescribeStack(&manager.Stack{StackName: aws.String(stackName)})
+	stack, err := a.stackManager.DescribeStack(ctx, &manager.Stack{StackName: aws.String(stackName)})
 	if err != nil {
-		if awsError, ok := errors.Unwrap(errors.Unwrap(err)).(awserr.Error); !ok || ok &&
-			awsError.Code() != "ValidationError" {
+		if awsError, ok := errors.Unwrap(errors.Unwrap(err)).(smithy.APIError); !ok || ok &&
```

Comment:
The stack not find error is not an APIError. It's an OperationError which doesn't implement APIError.

This path not unit tested.

Reply:
Does this error checking make sense? I mean, it returns not found if the stack doesn't exist. But we also check if the stack length is zero, which is again a different error, but it's the same. This is some weird code here. :D